src/Avalonia.Input/MouseDevice.cs

```diff
@@ -196,14 +168,15 @@ private void ProcessRawEvent(RawPointerEventArgs e)
             }
         }
 
-        private void LeaveWindow(IMouseDevice device, ulong timestamp, IInputRoot root, PointerPointProperties properties,
+        private void LeaveWindow(IMouseDevice device, ulong timestamp, IInputRoot root, PixelPoint? position, PointerPointProperties properties,
             KeyModifiers inputModifiers)
         {
             device = device ?? throw new ArgumentNullException(nameof(device));
             root = root ?? throw new ArgumentNullException(nameof(root));
 
+            var oldPosition = position.HasValue ? root.PointToClient(position.Value) : new Point(-1, -1);
```

Comment:
Previously mouse leave was generating leave-args without position information. For mouse it makes sense to pass previous known position before pointer left window boundaries.